### ISSUE TITLE
Rule option proposal: ignore void returns or no-return functions for `explicit-function-return-type`

### ISSUE THREAD

**Issue by Samuel-Therrien-Beslogic** [typescript-eslint/typescript-eslint#4826]
### Description

Right now, I have to write the following with this rule (see `Pass` for how I'd like to write this). Even with all `allow*` options turned on.
```ts
const listener: (event: ClipboardEvent, link: string) => void = (event, link) => {
  event.clipboardData?.setData('text/plain', link);
  event.preventDefault();
};
```

### Fail

```ts
const listener = (event: ClipboardEvent, link: string) => {
  event.clipboardData?.setData('text/plain', link);
  event.preventDefault();
  return true;
};
```

### Pass

```ts
const listener = (event: ClipboardEvent, link: string) => {
  event.clipboardData?.setData('text/plain', link);
  event.preventDefault();
};
```

**Comment by bradzacher** [comment]
Please use the search next time: #3534, #50
